pyqtgraph/functions.py

```diff
@@ -1794,9 +1800,52 @@ def arrayToQPath(x, y, connect='all'):
 
     ds = QtCore.QDataStream(buf)
     ds >> path
-
     return path
 
+def array2d_to_qpolygonf(xdata, ydata):
+    """
+    Utility function to convert two 1D-NumPy arrays representing curve data
+    (X-axis, Y-axis data) into a single open polygon (QtGui.PolygonF) object.
+
+    Thanks to PythonQwt for making this code available
+
+    License/copyright: MIT License © Pierre Raybaut 2020.
+
+    :param numpy.ndarray xdata: 1D-NumPy array (numpy.float64)
+    :param numpy.ndarray ydata: 1D-NumPy array (numpy.float64)
+    :return: Polyline
+    :rtype: QtGui.QPolygonF
+    """
+    if not (
+        xdata.size == ydata.size == xdata.shape[0] == ydata.shape[0]
+        and np.issubdtype(xdata.dtype, np.double)
+        and np.issubdtype(ydata.dtype, np.double)
+    ):
+        raise ValueError("Arguments must be 1D, float64 NumPy arrays with same size")
+    size = xdata.size
+    dtype = xdata.dtype
+    if QtVersion.startswith("5"):
+        polyline = QtGui.QPolygonF(size)
+    else:
+        if QT_LIB == "PySide6":
+            polyline = QtGui.QPolygonF()
+            polyline.resize(size)
+        else:
+            polyline = QtGui.QPolygonF([QtCore.QPointF(xdata[0], ydata[0])] * size)
+    if QT_LIB == "PySide2":
```

Comment:
Slightly less round-about way that avoids the trip to a ctypes object.
```python
nbytes = 2 * size * np.finfo(dtype).dtype.itemsize
if QT_LIB == "PySide2":
    buffer = Qt.shiboken2.VoidPtr(polyline.data(), nbytes, True)
elif QT_LIB = "PySide6":
    buffer = Qt.shiboken6.VoidPtr(polyline.data(), nbytes, True)
else:
    buffer = polyline.data()
    buffer.setsize(nbytes)
```

Reply:
that is much nicer; I really should spend a little time learning about the shiboken methods and how they work (and the pyqt5/6-sip ones for that matter too).